### DISCUSSION TITLE
Update state with URL hash.

### DISCUSSION THREAD

**Discussion by joel-jo-querypie** [pmndrs/zustand#1251] [Ideas] [answered]
Hello, Thank you for running a cute and powerful project.

I think it is useful to update state with URL Hash.
If you agree with this opinion, is it okay for me to work on it?

ex)
```js
const bears = useBearStore((state) => state.bears)
const increasePopulation = useBearStore((state) => state.increasePopulation)

function BearCounter() {
  const [bearHash, updateBearHash] = hashState('bears', bears, increasePopulation, 1); // key, state, updater, initailValue

  return <button onClick={updateBearHash}>one up</button>
}
```
click button > update to url to `~/#bears=2`

Thank you. Have a nice day :)

**Comment by dai-shi** [comment, marked as answer]
You should be able to use `persist` middleware which accepts custom storage. Please give it a try. If it's successful, it would be nice for docs.
(I did something similar in jotai's atomWithHash.)

**Reply by joel-jo-querypie** [reply]
Thank you for sharing the good content. I think it is possible to make a hashStorage that stores state with Url Hash. 
Thanks. I'll try this and then contribute to docs!

**Reply by joel-jo-querypie** [reply]
Is it okay to make "Document for state with URL Hash" an issue?

**Reply by dai-shi** [reply]
Sure.
